src/test/suite/tokenizer.test.ts

```diff
@@ -0,0 +1,58 @@
+import * as assert from "assert";
+import { tokenize } from "../../tokenizer";
+import { flatten, range } from "lodash";
+
+type TestType = [string, string[]][];
+const singleSymbolTests: TestType = getAsciiSymbols().map((s) => [s, [s]]);
+
+const tests: TestType = [
+  // Numbers
+  ["0.0 0 1 120 2.5 0.1", ["0.0", "0", "1", "120", "2.5", "0.1"]],
+  // Semantic versioning
+  ["1.22.4", ["1", ".", "22", ".", "4"]],
+  // Formatters and identifiers
+  ["my variable", ["my", "variable"]],
+  ["My Variable", ["My", "Variable"]],
+  ["myVariable", ["myVariable"]],
+  ["MyVariable", ["MyVariable"]],
+  ["my_variable", ["my_variable"]],
+  ["MY_VARIABLE", ["MY_VARIABLE"]],
+  ["my__variable", ["my__variable"]],
+  ["my-variable", ["my", "-", "variable"]],
+  ["my.variable", ["my", ".", "variable"]],
+  ["my/variable", ["my", "/", "variable"]],
+  ["my::variable", ["my", ":", ":", "variable"]],
+  // Strings
+  ['"my variable"', ['"', "my", "variable", '"']],
+  ["'my variable'", ["'", "my", "variable", "'"]],
+  // Symbols incl repeatable
+  ...singleSymbolTests,
+  ["!_|||>{.-----()", ["!", "_", "|||", ">", "{", ".", "-----", "(", ")"]],
```

Comment:
Nice. Would be good to have multiple parens in a row I mean, eg `(((` and `)))` because I prefer these to be split so want to make sure they don't end up in repeated symbols regex by accident